webapp/views.py

```diff
@@ -674,6 +674,71 @@ def tutorials_index():
     return tutorials_index
 
 
+def build_engage_index(engage_docs):
+    def engage_index():
+        page = flask.request.args.get("page", default=1, type=int)
+        topic = flask.request.args.get("topic", default=None, type=str)
+        sort = flask.request.args.get("sort", default=None, type=str)
+        posts_per_page = 15
+        engage_docs.parser.parse()
+        metadata = engage_docs.parser.metadata
+
+        total_pages = math.ceil(len(metadata) / posts_per_page)
+
+        return flask.render_template(
+            "engage/index.html",
+            forum_url=engage_docs.parser.api.base_url,
+            metadata=metadata,
+            page=page,
+            topic=topic,
+            sort=sort,
+            posts_per_page=posts_per_page,
+            total_pages=total_pages,
+        )
+
+    return engage_index
+
+
+def engage_thank_you(engage_pages):
+    """
+    Renders an engage pages thank-you page
+    i.e. whitepapers, pdfs
+
+    If there is no current topic it can't render the page
+    e.g. accessing directly
+
+    @parameter: an EngagePages instance
+    @returns: a function that renders a template
+        - If it does not come from a topic, abort
+        - If it comes from a topic, render the page
+        with the data
+    """
+
+    def render_template():
+
+        if hasattr(engage_pages.parser, "current_topic"):
```

Comment:
I'm not entirely convinced we should be rendering a template based on the state of a property of a `global` object.
Can we just  pass the `topic_id` as a parameter to the thank you view and get the info in the view?

Reply:
That will change with the new way we are rendering the thank-you page. I will update it.